runtime/src/main/java/org/corfudb/runtime/object/transactions/OptimisticTransactionalContext.java

```diff
@@ -56,6 +49,13 @@
     private final Set<ICorfuSMRProxyInternal> modifiedProxies =
             new HashSet<>();
 
+    /** Keep the version (at global snapshot time)
```

Comment:
What I am trying to achieve is to keep the version number of the underlying object at the global snapshot time. Upon access (after a context switch), I compare if the underlying object is at the same version number than where we started from in this context (the version of the object at the global snapshot).

Since we always compare the global snapshot against the version of the object, they will almost never be the same (global vs stream tail). The first time we access the object in the context, we sync the object (readto to the global tail) and we store the stream tail in the cache.

Next time we context switch, if this version number is the same in both context, we don't sync the object, we only perform optimistic rollbacks if any.

The confusion is in my language, I should have said version number, not version. Will change it.